### ISSUE TITLE
config.h in cmake build doesn't define LLVM_VERSION

### ISSUE THREAD

**Issue by dsandersimgtec** [pocl/pocl#159]
When configuring with:
```shell
cmake -GNinja ../pocl.src/
```

cmake correctly detects the LLVM version number.
```
-- LLVM_VERSION: 3.6
```

but the config.h does not contain any reference to LLVM_VERSION.

As a result it fails with:
```
[53/394] Building C object lib/CL/CMakeFiles/libpocl_unlinked_objs.dir/clBuildProgram.c.o
FAILED: /usr/bin/cc   -fPIC -I/mnt/scratch/das/pocl/pocl.src/lib/CL/../../fix-include/OpenCL -I. -I/mnt/scratch/das/pocl/pocl.src/fix-includes -I/mnt/scratch/das/pocl/pocl.src/include -I/mnt/scratch/das/pocl/pocl.src/lib/CL/devices -I/mnt/scratch/das/pocl/pocl.src/lib/CL/. -I/mnt/scratch/das/llvm/release_36/testing/clang+llvm-3.6.0-mipsel-linux-gnu/include -I/mnt/scratch/das/llvm/release_36/release/final/llvm.src/tools/clang/include -I/mnt/scratch/das/llvm/release_36/testing/clang+llvm-3.6.0-mipsel-linux-gnu/tools/clang/include    -I/usr/local/include -MMD -MT lib/CL/CMakeFiles/libpocl_unlinked_objs.dir/clBuildProgram.c.o -MF lib/CL/CMakeFiles/libpocl_unlinked_objs.dir/clBuildProgram.c.o.d -o lib/CL/CMakeFiles/libpocl_unlinked_objs.dir/clBuildProgram.c.o   -c /mnt/scratch/das/pocl/pocl.src/lib/CL/clBuildProgram.c
/mnt/scratch/das/pocl/pocl.src/lib/CL/clBuildProgram.c: In function ‘build_program_compute_hash’:
/mnt/scratch/das/pocl/pocl.src/lib/CL/clBuildProgram.c:100:43: error: ‘LLVM_VERSION’ undeclared (first use in this function)
   pocl_SHA1_Update (&hash_ctx, (uint8_t*) LLVM_VERSION, 
                                           ^
/mnt/scratch/das/pocl/pocl.src/lib/CL/clBuildProgram.c:100:43: note: each undeclared identifier is reported only once for each function it appears in
[53/394] Building CXX object lib/llvmopencl/CMakeFiles/llvmpasses.dir/linker.cpp.o
ninja: build stopped: subcommand failed.
```

The same thing happens with
```
cmake -GNinja ../pocl.src/
```
The autotools build seems to be ok though.

**Comment by dsandersimgtec** [comment]
The second cmake command should be:
```
cmake ../pocl.src/
```

**Comment by pjaaskel** [comment]
Fixed in 2fa3861334212fc
